### PR TITLE
Fauxpose

### PR DESCRIPTION
**Pull request by wsherman64**
This adds the "FauxPose" plugin which generates tracking pose data based on the mathematical expression of a circle, with the Y-value held constant, and with the orientation of the HMD facing forward.  This provides a utilities whereby users of IlliXR can test the basic software environment prior to having a working SLAM or other tracking system.

A sample "faux.yaml" configuration file is also provided to allow quick testing of the plugin.

### CODE REVIEW

**Review by shg8 — APPROVED** [review]

File: configs/faux.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# This IlliXR YAML configuration makes use of the FauxPose plugin, and thus
+#   does not require any of the SLAM tracking technologies to have been
+#   built on the system -- making it easier to test.
+plugin_groups:
+  #- !include "rt_slam_plugins.yaml"
+  - plugin_group:
+      #- path: ground_truth_slam/
+      - path: FauxPose/
```
Comment (Boyuan-Tian) [inline configs/faux.yaml:8]: Might want to use lowercase to be consistent with other plugins.

File: configs/faux.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# This IlliXR YAML configuration makes use of the FauxPose plugin, and thus
```
Comment (Boyuan-Tian) [inline configs/faux.yaml:1]: Should use `ILLIXR` as a consistent notion, similar for other places.

File: FauxPose/plugin.cpp [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+        // Adjust parameters based on environment variables
+        if ((env_input = getenv("FAUXPOSE_PERIOD"))) {
+            period = atof(env_input);
+        }
+        if ((env_input = getenv("FAUXPOSE_AMPLITUDE"))) {
+            amplitude = atof(env_input);
+        }
+        if ((env_input = getenv("FAUXPOSE_CENTER"))) {
+            // strchrnul() amplitude = atof(env_input);
```
Comment (Boyuan-Tian) [inline FauxPose/plugin.cpp:69]: Could remove this if unnecessary.

File: FauxPose/plugin.cpp [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+        return false;
+    }
+
+    // ********************************************************************
+    virtual pose_type correct_pose([[maybe_unused]] const pose_type pose) const override {
+        pose_type simulated_pose;
+#ifndef NDEBUG
+        std::cout << "[fauxpose] Returning (zero) pose\n";
+#endif
+        return simulated_pose;
```
Comment (jianxiapyh) [inline FauxPose/plugin.cpp:111]: why not just return pose_type{};?

**Comment by Jebbly** [comment]
Closed in favor of #389

**Pull request closed without merging**